src/mlpack/methods/cf/cf_main.cpp

```diff
@@ -125,10 +134,20 @@ PARAM_INT_IN("recommendations", "Number of recommendations to generate for each"
 
 PARAM_INT_IN("seed", "Set the random seed (0 uses std::time(NULL)).", "s", 0);
 
+//  Interpolation and Neighbour Search Algorithms
+PARAM_STRING_IN("interpolation", "Algorithm used for weight interpolation.",
+    "i", "Average");
+
+PARAM_STRING_IN("neighbour_search", "Algorithm used for neighbour search.",
+    "f", "Euclidean");
+
 void ComputeRecommendations(CFModel* cf,
                             const size_t numRecs,
                             arma::Mat<size_t>& recommendations)
 {
+  const string ns_algo = CLI::GetParam<string>("neighbour_search");
```

Comment:
Picky comment, do you think we should use a more descriptive name something like ``neighbourSearchPolicy`` instead of ``ns_algo``. Let me know what you think.